crates/sui-faucet/src/faucet/simple_faucet.rs

```diff
@@ -548,6 +572,251 @@ impl SimpleFaucet {
+        let pt = {
+            let mut builder = ProgrammableTransactionBuilder::new();
+            for (_uuid, recipient, amounts) in batch_requests {
+                let recipients: Vec<SuiAddress> =
+                    std::iter::repeat(recipient).take(amounts.len()).collect();
+                builder.pay_sui(recipients, amounts)?;
+            }
+            builder.finish()
+        };
+
+        Ok(TransactionData::new_programmable(
+            signer,
+            vec![gas_payment],
+            pt,
+            budget,
+            gas_price,
+        ))
+    }
+
+    async fn batch_gather(
+        &self,
+        requests: &mut Vec<(Uuid, SuiAddress, Vec<u64>)>,
+    ) -> Result<(), FaucetError> {
+        let Ok(mut req_consumer) = tokio::time::timeout(LOCK_TIMEOUT, self.request_consumer.lock()).await else {
+            error!("Timeout when getting request consumer lock");
+            return Err(FaucetError::Internal("Unable to get request queue lock.".to_string()));
+        };
+
+        let mut counter = 0;
+        // We take the first 500 items off the queue.
+        while counter < self.batch_request_size - 1 {
+            // We do need a timeout here or else it waits forever and gets stuck in this loop
+            let request = req_consumer.recv().await;
+            match request {
+                Some(req) => {
+                    requests.push(req);
+                    counter += 1;
+                }
+                None => {
+                    // Fail with error logging, so that we can continue, but we should never reach here.
+                    error!("Empty request stored in request consumer queue.");
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    // Function to process the batch send of the mcsp queue
+    pub async fn batch_transfer_gases(&self) -> Result<TransactionDigest, FaucetError> {
+        let Ok(mut req_consumer) = tokio::time::timeout(LOCK_TIMEOUT, self.request_consumer.lock()).await else {
+                error!("Timeout when getting request consumer lock");
+                return Err(FaucetError::Internal("Unable to get request queue lock.".to_string()));
+            };
+
+        let mut requests = Vec::new();
+
+        // First request we just wait for without timeout
+        let first_req = req_consumer.recv().await;
+        match first_req {
+            Some((uuid, address, amounts)) => {
+                requests.push((uuid, address, amounts));
+            }
+            None => {
+                return Err(FaucetError::Internal(
+                    "Request consumer queue has corrupted data.".to_string(),
+                ));
+            }
+        }
+        drop(req_consumer);
+        match timeout(BATCH_TIMEOUT, self.batch_gather(&mut requests)).await {
+            Ok(_) => {}
+            Err(_) => {
+                warn!("Batch timeout elapsed while waiting.");
+            }
+        };
```

Comment:
`warn` seems like too high a log level for this -- this should happen regularly when the system is not under load.

```suggestion
        if timeout(BATCH_TIMEOUT, self.batch_gather(&mut requests)).await.is_err() {
            debug!("Batch timeout elapsed while waiting.");
        };
```